### PR TITLE
Add RLS policy template to "Enable users to view their own data only"

### PR DESCRIPTION
**Pull request by tristanbob**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES, however, I am not able to test this change because the `npm install` process failed.

## What kind of change does this PR introduce?

New policy template

## What is the current behavior?

Currently, the PostgreSQL policy templates in the getGeneralPolicyTemplates function provide a variety of access control scenarios, but they do not include a policy specifically designed to restrict users to viewing only their own data.

## What is the new behavior?

A new policy template has been added that enables users to view only their own data. This policy ensures that users can only access rows where the `user_id` or `created_by` column matches their own user ID.

### New Policy:
- Template Name: Enable users to view their own data only
- Description: Grants read access (SELECT) to users so they can only view rows where the user_id or created_by column matches their own user ID.
- SQL Statement:
```
create policy "Enable users to view their own data only"
on "${schema}"."${table}"
for select
to authenticated
using (auth.uid() = user_id OR auth.uid() = created_by);
```

## Additional context

This new policy is crucial for applications that require data privacy and security in multi-user environments. It helps enforce row-level security by ensuring users can only access their own records, thus protecting sensitive data.

Thanks to @saltcod for suggesting this PR.

### CODE REVIEW

**Comment by charislam** [comment]
Thanks @tristanbob! Pushed a commit to tweak the SQL for the policy. I'll wait for a second reviewer to approve since I touched the code 😄

**Review by saltcod — APPROVED** [review]

**Comment by saltcod** [comment]
Great, thanks @tristanbob !!

**Pull request merged**
